// node-proxy-http/lib/options.js
'use strict';

Object.defineProperty(exports, "__esModule", {
    value: true
});

var _url = require('url');

var _url2 = _interopRequireDefault(_url);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var defaultHost = 'http://localhost';
var getQuery = function getQuery(originalUrl) {
    var primaryParamKey = arguments.length > 1 && arguments[1] !== undefined ? arguments[1] : 'q';

    var index = originalUrl.indexOf(primaryParamKey + '=');
    return originalUrl.substring(index + 2);
};

exports.default = function () {
    var setting = arguments.length > 0 && arguments[0] !== undefined ? arguments[0] : {};
    return {
        target: defaultHost,
        router: function router(req) {
            var q = getQuery(req.originalUrl);
            var parsedUrl = _url2.default.parse(q);
            var target = parsedUrl.protocol + '//' + parsedUrl.host;
            console.log('Target: ' + target);
            return target;
        },
        pathRewrite: function pathRewrite(path, req) {
            var q = getQuery(req.originalUrl, setting.key);
            var parsedUrl = _url2.default.parse(q);
            var urlPath = parsedUrl.path;
            console.log('Path: ' + urlPath);
            return urlPath;
        },
        onProxyRes: function onProxyRes(proxyRes, req, res) {
            var q = getQuery(req.originalUrl);
            console.log('Proxying to ' + q);
            console.log('------');
        },
        onProxyReq: function onProxyReq(proxyReq, req, res) {
            proxyReq.headers = req.headers;
        },
        changeOrigin: true
    };
};
